classes/task/get_meeting_reports.php

```diff
@@ -246,7 +254,12 @@ public function format_participant($participant, $detailsid, $names, $emails) {
         }
 
         if ($participant->user_email == '') {
-            $participant->user_email = null;
+            if (!empty($moodleuserid)) {
+                $participant->user_email = $DB->get_record('user',
+                                        ['id' => $moodleuserid], '*', IGNORE_MULTIPLE)->email;
```

Comment:
id is unique, so we shouldn't need the two optional parameters.
```suggestion
                $participant->user_email = $DB->get_record('user', ['id' => $moodleuserid])->email;
```

However, that doesn't guarantee that it will match a row in the table. So you might need to check the a row was returned.